core/templates/domain/exploration/StatesObjectFactory.ts

```diff
@@ -155,6 +167,33 @@ export class States {
     }
     return allAudioTranslations;
   }
+
+  getTotalRequiredWrittenTranslationsCount(): number {
+    let total = 0;
+    Object.values(this._states).forEach(
+      state => total += state.getRequiredWrittenTranslationsCount());
+    return total;
+  }
+
+  getAllWrittenTranslations(
+      languageCode: string
+  ): WrittenTranslationObjectsDict {
+    const allWrittenTranslations = {};
+    for (const stateName in this._states) {
+      const state = this._states[stateName];
+      allWrittenTranslations[stateName] = [];
+      const contentIdsList = state.writtenTranslations.getAllContentId();
+      contentIdsList.forEach(contentId => {
+        const writtenTranslation = (
+          state.writtenTranslations.getWrittenTranslation(
```

Comment:
If I'm not mistaken, you get all written translations just to count them. If that's the case, why not have this method do the counting instead? The approach of "iterate through everything, try fetching the translation, and ignore if undefined" is going to be unnecessarily time-consuming for languages with few bits translated, right?

Also, if you do it that way, you could break early once you see more than 5 bad ones.

I think this _might_ be a real concern and not just a nice-to-have, because I tried loading a typical lesson and it took quite a long time to display (perhaps because it was doing this computation).